collector/internal/lifecycle/manager.go

```diff
@@ -107,12 +101,18 @@ func (lm *manager) Run(ctx context.Context) error {
 		return err
 	}
 
+	lm.wg.Add(1)
```

Comment:
What was the reason waitgroup Add was moved out from `processEvents` to here?

Reply:
We need to move that line out because we need to set the wait group before kicking out the go routine. Otherwise, there will be another race condition between `processEvents`(setting the wait group & `lm.wg.Wait()`

Reply:
I see, tricky.